### PR TITLE
update drupal_user.rb to update status of User record

### LINKED ISSUE #2156: Update user.status whenever drupal_user.status changes
### Please describe the problem (or idea)

Currently user.status is not updated whenever a user is banned or unbanned. The change in status is required for user.status too. 
The problem was discussed in pr #2047 

The following things need to be taken care of:-
1) A migration to sync user status with drupal_user status
2) Set user.status = 1 by default when new user is created (currently user status sets to 0)
3) refactor all ban and unban methods in drupal user to update user.status along with drupal_user.status

### What's your PublicLab.org username?

vidit

## Thank you!

Your help makes Public Lab better! We *deeply* appreciate your helping refine and improve this site. 

To learn how to write really great issues, which increases the chances they'll be resolved, see:

https://publiclab.org/wiki/developers#Contributing+for+non-coders

### PR DESCRIPTION
**Pull request by ViditChitkara**
fixes #2156

Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

File: db/migrate/20180128162224_sync_user_status_with_drupal_user_status.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class SyncUserStatusWithDrupalUserStatus < ActiveRecord::Migration
+  def up
+	  DrupalUser.all.each do |du|
```
Comment (jywarren) [inline db/migrate/20180128162224_sync_user_status_with_drupal_user_status.rb]: Hi! This looks awesome, thanks! Could you use 2-space indents here, though? Sorry to be so formatting inflexible :-)

File: app/models/drupal_user.rb [hunk 2]
```diff
@@ -69,13 +72,20 @@ def unban
...
+    update_user_status(1)
     self
   end
 
   def email
     mail
   end
 
+  def update_user_status(status)
+    u = user
```
Comment (jywarren) [inline app/models/drupal_user.rb]: Could we make this `self.user` just to clarify a bit?

Comment (ViditChitkara) [reply app/models/drupal_user.rb]: Sure thing!!

File: db/migrate/20180128162224_sync_user_status_with_drupal_user_status.rb (hunk 1 above)
Comment (ViditChitkara) [reply db/migrate/20180128162224_sync_user_status_with_drupal_user_status.rb]: Yeah, I noticed that. Don't know why but I had 2 space indent in my text Editor and seems to distort here. No worries, I'll try and correct it!

**Comment by ViditChitkara** [comment]
Made the required changes. @jywarren please have a look!

**Comment by jywarren** [comment]
Hooray. This is the best. Thanks so much!!!!

🎉 👍 👍

**Pull request merged**